indico_toolkit/snapshots/snapshot.py

```diff
@@ -0,0 +1,210 @@
+            path_to_snapshot (str): path to Snapshot CSV
+            text_col (str, optional): Column with text, will be inferred if not provided. Defaults to None.
+            label_col (str, optional): Column with labels, will be inferred if not provided. Defaults to None.
+            file_name_col (str, optional): Column with file names, will be inferred if not provided. Defaults to None.
+        """
+        self.path_to_snapshot = path_to_snapshot
+        self.df: pd.DataFrame = pd.read_csv(path_to_snapshot)
+        self.label_col = label_col
+        if label_col is None:
+            self._infer_label_col()
+        self._convert_col_from_json()
+        self.text_col = text_col
+        if text_col is None:
+            self._infer_text_col()
+        self.file_name_col = file_name_col
+        if file_name_col is None:
+            self._infer_file_name_col()
+
+    def remove_extraction_labels(self, labels_to_remove: List[str]):
+        """
+        Remove unwanted labels from label column
+        Args:
+            labels_to_remove (List[str]): the labels you'd like to remove
+        """
+        updated_predictions = []
+        for label_set in self.df[self.label_col]:
+            updated_predictions.append(
+                [i for i in label_set if i["label"] not in labels_to_remove]
+            )
+        self.df[self.label_col] = updated_predictions
+
+    def standardize_column_names(
+        self,
+        text_col: str = "source",
+        label_col: str = "target",
+        file_name_col: str = "file_name",
+    ):
+        """
+        To allow merging/appending snapshots, standardize the key column headers
+        """
+        self.df.rename(
+            mapper={
+                self.file_name_col: file_name_col,
+                self.text_col: text_col,
+                self.label_col: label_col,
+            },
+            inplace=True,
+            errors="raise",
+            axis=1,
+        )
+        self.file_name_col = file_name_col
+        self.label_col = label_col
+        self.text_col = text_col
+
+    def drop_unneeded_columns(self, columns_to_drop: List[str] = None):
+        """
+        Keep only text, labels, and file name columns or specify columns to drop by passing them in as a list.
+        """
+        if columns_to_drop:
+            self.df.drop(labels=columns_to_drop, axis=1, inplace=True)
+        else:
+            self.df = self.df[[self.label_col, self.text_col, self.file_name_col]]
+
+    def append(self, snap_to_add: Snapshot):
+        """
+        Append the rows from another Snapshot to this snapshot. Ensure column names are standardized beforehand.
+        Args:
+            snap_to_add (Snapshot): Snapshot to add
+        """
+        self._assert_key_column_names_match(snap_to_add)
+        self.df = self.df.append(snap_to_add.df, ignore_index=True)
+
+    def merge_by_file_name(
+        self, snap_to_merge: Snapshot, ensure_identical_text: bool = True,
```

Comment:
could add inplace arg and return a new snapshot object, but might not be useful.

Reply:
I think it could be, but I stripped all of the 'inplace' kwargs out of Association to try to enforce OOP style where we control how it gets written, so'll leave this without for now, but down to make that change later